src/org/jitsi/impl/neomedia/ByteArrayBufferImpl.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia;
+
+import org.jitsi.service.neomedia.*;
+
+import java.util.*;
+
+/**
+ * Implements {@link ByteArrayBuffer}.
+ * @author Boris Grozev
+ */
+public class ByteArrayBufferImpl
+    implements ByteArrayBuffer
+{
+    /**
+     * The byte array represented by this {@link ByteArrayBufferImpl}.
+     */
+    private byte[] buffer;
+
+    /**
+     * The offset in the byte buffer where the actual data starts.
+     */
+    private int offset;
+
+    /**
+     * The length of the data in the buffer.
+     */
+    private int length;
+
+    /**
+     * Initializes a new {@link ByteArrayBufferImpl} instance.
+     * @param buffer
+     * @param offset
+     * @param length
+     */
+    public ByteArrayBufferImpl(byte[] buffer, int offset, int length)
+    {
+        this.buffer = Objects.requireNonNull(buffer, "buffer");
+        if (offset + length > buffer.length)
```

Comment:
I expect this to be a commonly used base class for specialized `ByteArrayBuffer`s. Could you please add the positive or 0 check that we discussed bellow?